### PR TITLE
Improve debugger window interactions

### PR DESCRIPTION
**Pull request by kcgen**
The PR attempts to prevent the debugger window from getting "stuck" on some platforms and SDL versions.

The PR also applies some non-functional cleanups, such as initializing global to default values and using range-based loops.

### CODE REVIEW

File: src/debug/debug.cpp [hunk 1]
```diff
@@ -1588,7 +1580,11 @@ int32_t DEBUG_Run(int32_t amount,bool quickexit) {
 	else {
 		// ensure all breakpoints are activated
 		CBreakpoint::ActivateBreakpoints();
-		SDL_RaiseWindow(GFX_GetSDLWindow());
+
+		const auto graphics_window = GFX_GetSDLWindow();
+		SDL_RaiseWindow(graphics_window);
+		SDL_SetWindowInputFocus(graphics_window);
```
Comment (kklobe) [inline src/debug/debug.cpp:1586]: `SDL_RaiseWindow` should already set input focus to the raised window, https://wiki.libsdl.org/SDL_RaiseWindow

Comment (kcgen) [reply src/debug/debug.cpp:1586]: Thanks @kklobe; will drop that.

Comment (kcgen) [reply src/debug/debug.cpp:1586]: @kklobe , can you try this:
 3. At the Z:\> prompt: `DEBUG.COM IMGMOUNT.COM`
 4. The debugger window will be raised automatically, no mouse action is needed.
 5. Tap F10 seven times to step instruction by instruction
 6. On seventh tap, the main gfx window is in foregrounded and (should!) be in-focus, press a keyboard key to confirm (again, no mouse action needed).

For me:
 - When the SDL_SetWindowInputFocus calls are commented out, step 6 doesn't focus my keyboard input after the 7th F10 tap.  The main GFX window is in the foreground, but keyboard input appears frozen. 
 - When the SDL_SetWindowInputFocus calls are left in, step 6 (for me) seamlessly transitions from the 7th F10 tap in the debugger window to now having focussed keyboard input in the main gfx window.  I can tap a keyboard key and it shows up on the Z:\> prompt.

Comment (kklobe) [reply src/debug/debug.cpp:1586]: the above sequence works just fine for me on `main`as well as on your branch - that's the behavior I intended and tested when I devved it. `main` also works on my Windows 10 / VS2022 setup.

Comment (kcgen) [reply src/debug/debug.cpp:1586]: Thanks @kklobe ; definitely know this was in good working order when your PRs landed :👍; very likely entirely Linux-specific. Will do more testing and also before and after the commit that @grapeli bisected in SDL2 (thank you!)

**Comment by kcgen** [comment]
Merging, as this is an incremental improvement per my own testing and feedback from #1821 ticket, given it doesn't make any structural changes to the PDCurses event queue or the OpenGL side.

I'll open a ticket and reproduction steps for the OpenGL issue.

**Pull request merged**